interfaces/builtin/cups.go

```diff
@@ -0,0 +1,62 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// On systems where the slot is provided by an app snap, the cups interface is
+// the companion interface to the cups-control interface. The design of these
+// interfaces is based on the idea that the slot implementation (eg cupsd) is
+// expected to query snapd to determine if the cups-control interface is
```

Comment:
> should that be "if the cups interface" here?

Actually, no. The idea is that since the socket is allowed in both interfaces, cupsd will see if 'cups-control' is connected to determine if admin functionality is allowed (cupsd can assume that since communications are happening at all that one of them is connected, so if cups-control isn't connected, only allow printing and not admin).